core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -112,74 +112,69 @@ def restart(self) -> bool:
 
     def _load_endpoints(self) -> None:
         """Load endpoints from the configuration file to the mavlink manager."""
-        if "endpoints" not in self.configuration:
-            self.configuration["endpoints"] = []
-        endpoints = self.configuration["endpoints"]
+        endpoints = {Endpoint(**endpoint) for endpoint in self.configuration.get("endpoints") or []}
```

Comment:
Isn't it better to have a `_get_configuration_endpoints()` call that gets and parse the configuration endpoint in a friendly way ?

Reply:
I didn't do it because that's the only interface between the configured endpoints and the APManager, but doing it does no harm, for sure.